mso/datasource_mso_site.go

```diff
@@ -86,67 +87,117 @@ func datasourceMSOSiteRead(d *schema.ResourceData, m interface{}) error {
 
 	msoClient := m.(*client.Client)
 	name := d.Get("name").(string)
-	con, err := msoClient.GetViaURL("api/v1/sites")
+	var path string
+	platform := msoClient.GetPlatform()
+	if platform == "nd" {
+		path = "api/v2/sites"
+	} else {
+		path = "api/v1/sites"
+	}
+	con, err := msoClient.GetViaURL(path)
 	if err != nil {
 		return err
 	}
 
 	data := con.S("sites").Data().([]interface{})
+
 	var flag bool
 	var count int
 
 	for _, info := range data {
-		val := info.(map[string]interface{})
-		if val["name"].(string) == name {
-			flag = true
-			break
+		if platform == "nd" {
+			val := info.(map[string]interface{})["common"]
+			if val.(map[string]interface{})["name"].(string) == name {
+				flag = true
+				break
+			}
+		} else {
+			val := info.(map[string]interface{})
+			if val["name"].(string) == name {
+				flag = true
+				break
+			}
 		}
+
 		count = count + 1
 	}
 
 	if flag != true {
 		return fmt.Errorf("Site of specified name not found")
 	}
 
-	dataCon := con.S("sites").Index(count)
-
+	var dataCon *container.Container
+	dataCon = con.S("sites").Index(count)
```

Comment:
can't you do dataCon := con.S("sites").Index(count) to replace those two lines?

Reply:
Yes!